modules/data-streams/src/main/java/org/elasticsearch/datastreams/lifecycle/DataStreamLifecycleService.java

```diff
@@ -1336,7 +1338,9 @@ static class ErrorRecordingActionListener implements ActionListener<Void> {
         @Override
         public void onResponse(Void unused) {
             logger.trace("Clearing recorded error for index [{}] because the [{}] action was successful", targetIndex, actionName);
-            errorStore.clearRecordedError(targetIndex);
+            @FixForMultiProject(description = "Don't use default project ID")
+            final var projectId = Metadata.DEFAULT_PROJECT_ID;
```

Comment:
Not for this PR. But I wonder whether we could remove `Metadata.DEFAULT_PROJECT_ID` and replace it with a deprecated method, e.g.:
```java
@Deprecated(forRemoval=true)
public static ProjectId defaultProjectId() {
    return ProjectId.DEFAULT;
}
```
We can then use the above method in all places where default project id is meant to be a placeholder. And use `ProjectId.DEFAULT` for places we actually want the default ID. What do you think?

Reply:
I think that's a great idea, we should have done that earlier :)